[html-elements] Move BR to Rules

sjchmiela opened the pull request:
# Why

`test-suite` was failing with an ominous error: **Views nested within a <Text> must have a width and height**.

# How

This took a while to debug (it seemed like the error was thrown while testing `Nav`, but in fact the logging was just slower than testing and it was thrown when testing `BR`). `test-suite` expects `BR` to be a view element, so I made it.

https://github.com/expo/expo/blob/90a1e11151ce0689ea0bf6d89b55cf37ceeff47b/apps/test-suite/tests/HTML.js#L74-L80

# Test Plan

Test no longer fails with **Views nested within…**.

EvanBacon commented on packages/html-elements/src/elements/Text.tsx:
```diff
@@ -34,9 +34,9 @@ export const BlockQuote = forwardRef(({ style, cite, ...props }: BlockQuoteProps
-export const BR = forwardRef(({ style, ...props }: TextProps, ref) => {
-  return <Text {...props} style={[styles.br, style]} ref={ref} />;
-}) as ComponentType<TextProps>;
+export const BR = forwardRef(({ style, ...props }: ViewProps, ref) => {
```
Should probably move this to the `Rules.tsx` files (same with the tests).

EvanBacon reviewed (changes requested):
Tests will need to be updated (`yarn test -u`)

sjchmiela committed:
[native-component-list] Add a visible BR test

sjchmiela committed:
[html-elements] Move BR to Rules and make it web-compatible

sjchmiela committed:
[html-elements] Update test snapshots

sjchmiela committed:
[test-suite] Test BR in a separate, special test case

sjchmiela commented:
- moved to `Rules`,
- reverted backing view to `Text`
- added a web-compatible implementation in `Rules.web.tsx` that uses native `<br />`
- added a special test case in `test-suite`, since neither `<Element>Test contents</Element>` nor `<Element><View /></Element>` applies to `<BR />` element.
- added a BR-ed paragraph to NCL to be able to test its functionality visually

sjchmiela commented:
Fixed by @esamelson in https://github.com/expo/expo/pull/8642.
